src/stream/mod.rs

```diff
@@ -79,7 +79,8 @@ pub struct Stream {
     pub(crate) closed: AtomicBool,
     pub(crate) unordered: AtomicBool,
     pub(crate) reliability_type: AtomicU8, //ReliabilityType,
-    pub(crate) reliability_value: AtomicU32,
+    pub(crate) has_reliability_value: AtomicBool,
+    pub(crate) reliability_value: AtomicU16,
```

Comment:
An alternative to two values is using `AtomicU32` and a sentinel value such as `u32::MAX` to mark an absent value. Splitting the value across two atomics does invite more raciness than a single value.